include/misc/stack.h

```diff
@@ -27,15 +28,30 @@ static inline size_t stack_unused_space_get(const char *stack, size_t size)
 	 * that correct Kconfig option is used.
 	 */
 #if defined(STACK_GROWS_UP)
-	for (i = size - 1; i >= 0; i--) {
+#if defined(CONFIG_STACK_SENTINEL)
+	if (*((u32_t*)&stack[size-4]) != 0xF0F0F0F0) {
```

Comment:
please don't use hard-coded sentinel value. we are checking for STACK_SENTINEL defined in kernel_structs.h. you can move the definition if its current location is inconvenient.

Reply:
actually I don't know why this is even being checked.
if the value is *not* present, then this particular stack has definitely overflowed and periodic calls to _check_stack_sentinel() should catch it.
why not just increment the value by 4?

Reply:
The change is not really to check for stack sentinel. I agree that if it is blown it will be caught elsewhere. The change is really so that the stack_unused_space_get() will report the correct value when someone has CONFIG_STACK_SENTINEL set.

Reply:
Where do you suggest I move STACK_SENTINEL?

Reply:
Would it be okay to conditionally include <kernel_structs.h> if sentinel checking is enabled?